pkg/converter/convert.go

```diff
@@ -80,91 +104,98 @@ func Convert(o Options) error {
-				}
-				if o.isSet(flagServerID) {
-					exec.Args = append(exec.Args, argServerID)
-					exec.Args = append(exec.Args, o.TokenOptions.ServerID)
-				} else if authInfo.AuthProvider.Config[cfgApiserverID] != "" {
-					exec.Args = append(exec.Args, argServerID)
-					exec.Args = append(exec.Args, authInfo.AuthProvider.Config[cfgApiserverID])
-				}
-				if o.isSet(flagClientID) {
-					exec.Args = append(exec.Args, argClientID)
-					exec.Args = append(exec.Args, o.TokenOptions.ClientID)
-				} else if !isAlternativeLogin && authInfo.AuthProvider.Config[cfgClientID] != "" {
-					// when MSI is enabled, the clientID in azure authInfo will be disregarded
-					exec.Args = append(exec.Args, argClientID)
-					exec.Args = append(exec.Args, authInfo.AuthProvider.Config[cfgClientID])
-				}
-				if !isAlternativeLogin && o.isSet(flagTenantID) {
-					exec.Args = append(exec.Args, argTenantID)
-					exec.Args = append(exec.Args, o.TokenOptions.TenantID)
-				} else if !isAlternativeLogin && authInfo.AuthProvider.Config[cfgTenantID] != "" {
-					exec.Args = append(exec.Args, argTenantID)
-					exec.Args = append(exec.Args, authInfo.AuthProvider.Config[cfgTenantID])
-				}
-				if !isAlternativeLogin && o.isSet(flagIsLegacy) && o.TokenOptions.IsLegacy {
-					exec.Args = append(exec.Args, argIsLegacy)
-				} else if !isAlternativeLogin && (authInfo.AuthProvider.Config[cfgConfigMode] == "" || authInfo.AuthProvider.Config[cfgConfigMode] == "0") {
-					exec.Args = append(exec.Args, argIsLegacy)
-				}
-				if !isAlternativeLogin && o.isSet(flagClientSecret) {
-					exec.Args = append(exec.Args, argClientSecret)
-					exec.Args = append(exec.Args, o.TokenOptions.ClientSecret)
-				}
-				if !isAlternativeLogin && o.isSet(flagClientCert) {
-					exec.Args = append(exec.Args, argClientCert)
-					exec.Args = append(exec.Args, o.TokenOptions.ClientCert)
-				}
-				if !isAlternativeLogin && o.isSet(flagUsername) {
-					exec.Args = append(exec.Args, argUsername)
-					exec.Args = append(exec.Args, o.TokenOptions.Username)
-				}
-				if !isAlternativeLogin && o.isSet(flagPassword) {
-					exec.Args = append(exec.Args, argPassword)
-					exec.Args = append(exec.Args, o.TokenOptions.Password)
-				}
-				if o.isSet(flagLoginMethod) {
-					exec.Args = append(exec.Args, argLoginMethod)
-					exec.Args = append(exec.Args, o.TokenOptions.LoginMethod)
+			if isAzureCLI { //support azurecli login for now
+				exec.Args = append(exec.Args, argServerID)
+				serveridArg := getServerId(authInfo)
+				if len(serveridArg) < 1 {
+					fmt.Fprintln(os.Stderr, "err: serverId")
+					os.Exit(1)
 				}
+				exec.Args = append(exec.Args, serveridArg)
+				exec.Args = append(exec.Args, argLoginMethod)
+				exec.Args = append(exec.Args, o.TokenOptions.LoginMethod)
 				authInfo.Exec = exec
 				authInfo.AuthProvider = nil
+				continue
+			} else {
+				// others are not supported yet
+				fmt.Fprintln(os.Stderr, "err: not supported yet")
```

Comment:
same here: return fmt.Errorf("%q is not supported yet", o.TokenOptions.LoginMethod)